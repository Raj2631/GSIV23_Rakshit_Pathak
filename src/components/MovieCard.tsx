import { Link } from "react-router-dom";
import { Movie } from "../services/types";
import { ImgBaseURL } from "../utils/constants";

type Props = {
  movie: Movie;
};

const MovieCard = ({ movie }: Props) => {
  const { title, overview, vote_average } = movie;
  return (
    <div className="w-52 rounded-lg cursor-pointer overflow-hidden shadow-[rgba(50,50,93,0.25)_0px_6px_12px_-2px,_rgba(0,0,0,0.3)_0px_3px_7px_-3px]">
      <Link to={`/movieDetail/${movie.id}`}>
        <img
          key={movie.id}
          src={`${ImgBaseURL}/${movie.poster_path}`}
          className="w-full bg-black bg-cover"
        />
        <div className=" py-2 px-2">
          <div className="flex items-center  justify-between text-sm">
            <div className="w-24 truncate text-left font-semibold">{title}</div>
            <div className="text-gray-400">({vote_average})</div>
          </div>
          <div className="w-full text-sm text-left my-1 text-gray-500 line-clamp-2">
            {overview}
          </div>
        </div>
      </Link>
    </div>
  );
};

export default MovieCard;
