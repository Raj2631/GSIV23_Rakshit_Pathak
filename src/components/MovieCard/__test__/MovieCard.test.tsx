import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import MovieCard from "../index";

const mockMovie = {
  adult: false,
  backdrop_path: "/jZIYaISP3GBSrVOPfrp98AMa8Ng.jpg",
  genre_ids: [16, 35, 10751, 14, 10749],
  id: 976573,
  original_language: "en",
  original_title: "Elemental",
  overview:
    "In a city where fire, water, land and air residents live together, a fiery young woman and a go-with-the-flow guy will discover something elemental: how much they have in common.",
  popularity: 4696.546,
  poster_path: "/6oH378KUfCEitzJkm07r97L0RsZ.jpg",
  release_date: "2023-06-14",
  title: "Elemental",
  video: false,
  vote_average: 7.81,
  vote_count: 1050,
};

describe("MovieCard Component", () => {
  it("renders correctly", () => {
    const tree = render(
      <Router>
        <MovieCard movie={mockMovie} />
      </Router>
    );

    const titleElement = tree.getByText(mockMovie.title);
    const overviewElement = tree.getByText(mockMovie.overview);
    const voteAverageElement = tree.getByText(`(${mockMovie.vote_average})`);
    const posterElement = tree.getByText(mockMovie.title);

    expect(titleElement).toBeInTheDocument();
    expect(overviewElement).toBeInTheDocument();
    expect(voteAverageElement).toBeInTheDocument();
    expect(posterElement).toBeInTheDocument();

    expect(tree).toMatchSnapshot();
  });

  it("displays default description when overview is missing", () => {
    const movieWithoutOverview = { ...mockMovie, overview: "" };

    const { getByText } = render(
      <Router>
        <MovieCard movie={movieWithoutOverview} />
      </Router>
    );

    const defaultDescriptionElement = getByText("No description available.");
    expect(defaultDescriptionElement).toBeInTheDocument();
  });
});
