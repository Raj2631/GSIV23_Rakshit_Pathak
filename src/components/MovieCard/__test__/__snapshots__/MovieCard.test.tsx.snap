// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieCard Component > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-60 h-[444px] rounded-lg cursor-pointer overflow-hidden shadow-[rgba(50,50,93,0.25)_0px_6px_12px_-2px,_rgba(0,0,0,0.3)_0px_3px_7px_-3px]"
      >
        <a
          href="/movieDetail/976573"
        >
          <img
            alt="Elemental"
            class="w-full h-[360px] bg-black bg-cover"
            src="https://image.tmdb.org/t/p/w500//6oH378KUfCEitzJkm07r97L0RsZ.jpg"
          />
          <div
            class=" py-2 px-2"
          >
            <div
              class="flex items-center  justify-between text-sm"
            >
              <div
                class="w-24 truncate text-left font-semibold"
              >
                Elemental
              </div>
              <div
                class="text-gray-400"
              >
                (
                7.81
                )
              </div>
            </div>
            <div
              class="w-full text-sm text-left my-1 text-gray-500 line-clamp-2"
            >
              In a city where fire, water, land and air residents live together, a fiery young woman and a go-with-the-flow guy will discover something elemental: how much they have in common.
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-60 h-[444px] rounded-lg cursor-pointer overflow-hidden shadow-[rgba(50,50,93,0.25)_0px_6px_12px_-2px,_rgba(0,0,0,0.3)_0px_3px_7px_-3px]"
    >
      <a
        href="/movieDetail/976573"
      >
        <img
          alt="Elemental"
          class="w-full h-[360px] bg-black bg-cover"
          src="https://image.tmdb.org/t/p/w500//6oH378KUfCEitzJkm07r97L0RsZ.jpg"
        />
        <div
          class=" py-2 px-2"
        >
          <div
            class="flex items-center  justify-between text-sm"
          >
            <div
              class="w-24 truncate text-left font-semibold"
            >
              Elemental
            </div>
            <div
              class="text-gray-400"
            >
              (
              7.81
              )
            </div>
          </div>
          <div
            class="w-full text-sm text-left my-1 text-gray-500 line-clamp-2"
          >
            In a city where fire, water, land and air residents live together, a fiery young woman and a go-with-the-flow guy will discover something elemental: how much they have in common.
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
